### DISCUSSION TITLE
Question about PX4 HITL

### DISCUSSION THREAD

**Discussion by mych907** [microsoft/AirSim#3467] [Support Q&A] [answered]
I'm a beginner at AirSim. During my study of AirSim with PX4, it seemed that a PX4 hardware could be connected to computer via USB. The PX4 can be used to control the movement of a multicopter in Unreal Engine, if I'm right.

Then my question is, what happens to the IMU data of the PX4 hardware? It is definitely stationary, connected to my computer. Shouldn't the sensor data conflict if the multicopter started moving? Or am I completely getting this wrong?

A little help to clear the confusion will be appreciated.

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @mych907 and welcome to the community ^_^ 
AFAIK (and according to [this doc](https://docs.px4.io/master/en/simulation/hitl.html)), when you select the HIL configuration/frame (Step 4 in the [AirSim HITL guide](https://microsoft.github.io/AirSim/px4_setup/#setting-up-px4-hardware-in-loop) similar to the image below) the 'real' sensors on the Pixhawk hardware are disabled. So moving the Pixhawk controller around would not results in anything (other than the USB cord disconnecting if its loose maybe 😂) since all the sensor data is coming from the simulator.

![image](https://user-images.githubusercontent.com/49626509/111184266-cc9ac280-857e-11eb-8756-ec8f57f2c643.png)

**Reply by mych907** [reply]
Amazing. I'm glad I asked here! Thanks for clarifying my confusion!!
